### DISCUSSION TITLE
User Defined Value Types?

### DISCUSSION THREAD

**Discussion by antoniopgs** [PaulRBerg/prb-math#124] [Q&A] [answered]
Will user defined value types be implemented?

**Comment by PaulRBerg** [comment, marked as answer]
YES! PRBMath V3 will come with user-defined value types. In fact, I have already implemented the in the `staging `branch, which can be used now, though I haven't shipped it formally yet because I have to finish writing all the tests.

Here's an example script:

```solidity
import { UD60x18 } from "@prb/math/UD60x18.sol";

function foo() pure returns (UD60x18 result) {
    UD60x18 x = UD60x18.wrap(0.09e18);
    UD60x18 y = UD60x18.wrap(2.71e18);
    result  = x.mul(y);
}
```

Also related: https://github.com/paulrberg/prb-math/pull/92

**Comment by PaulRBerg** [comment]
**Update** for future visitors: PRBMath [v3.0.0](https://github.com/paulrberg/prb-math/releases/tag/v3.0.0) is now live, which means that PRBMath has support for user defined value types 🥳

@antoniopgs I would absolutely love your feedback on V3!

**Reply by antoniopgs** [reply]
Having trouble installing this on forge.
Currently troubleshooting on the Foundry Support Chat.
You might want to check it out. https://web.telegram.org/k/#@foundry_support

**Reply by antoniopgs** [reply]
Will also open issue.

**Reply by PaulRBerg** [reply]
Thanks @antoniopgs, I replied to that issue.
